--- midori/midori-websettings.c.orig	2008-10-09 06:15:39.000000000 +0300
+++ midori/midori-websettings.c	2008-10-09 06:16:09.000000000 +0300
@@ -401,7 +401,7 @@
                                      "homepage",
                                      _("Homepage"),
                                      _("The homepage"),
-                                     "http://www.google.com",
+                                     "http://www.pardus.org.tr",
                                      flags));
 
     g_object_class_install_property (gobject_class,
