diff -Nuar elsa.orig//configure.ac elsa//configure.ac
--- elsa.orig//configure.ac	2011-02-24 16:33:02.000000000 +0200
+++ elsa//configure.ac	2011-03-05 23:05:08.865000034 +0200
@@ -46,6 +46,20 @@
 AC_MSG_CHECKING([whether to support pam])
 AC_MSG_RESULT([${enable_pam}])
 
+# ConsoleKit support
+AC_ARG_WITH(consolekit, AC_HELP_STRING([--with-consolekit], [Use ConsoleKit]),
+   [USE_CONSOLEKIT=$withval], [USE_CONSOLEKIT=yes])
+if test x"$USE_CONSOLEKIT" != xno; then
+   PKG_CHECK_MODULES(CK_CONNECTOR, ck-connector,
+       [USE_CONSOLEKIT=yes], [USE_CONSOLEKIT=no])
+   if test x"$USE_CONSOLEKIT" = xyes; then
+       AC_DEFINE([USE_CONSOLEKIT], 1, [Define to 1 to use ConsoleKit])
+       CFLAGS="$XDM_CFLAGS $CK_CONNECTOR_CFLAGS"
+       LIBS="$XDM_LIBS $CK_CONNECTOR_LIBS"
+   fi
+fi
+dnl AM_CONDITIONAL(USE_CONSOLEKIT, test$USE_CONSOLEKIT = xyes)
+
 # Set edje_cc path
 AC_ARG_WITH([edje-cc],
    [AC_HELP_STRING([ --with-edje-cc=PATH], [specify a specific path to edje_cc])],
diff -Nuar elsa.orig//src/daemon/elsa_session.c elsa//src/daemon/elsa_session.c
--- elsa.orig//src/daemon/elsa_session.c	2011-02-24 16:33:00.000000000 +0200
+++ elsa//src/daemon/elsa_session.c	2011-03-05 23:06:56.764000034 +0200
@@ -8,7 +8,8 @@
 #include <stdlib.h>
 #include <sys/types.h>
 #include <sys/wait.h>
-
+#include <ConsoleKit/ck-connector/ck-connector.h>
+#include <glib.h>
 #include "elsa.h"
 
 
@@ -19,6 +20,10 @@
 static pid_t _session_pid;
 static int _elsa_session_userid_set(struct passwd *pwd);
 static void _elsa_session_run(struct passwd *pwd, const char *cmd, const char *cookie);
+static CkConnector *ckc;
+static int tty = 7;
+GKeyFile *config;
+
 
 long
 elsa_session_seed_get()
@@ -108,6 +113,30 @@
 #ifdef HAVE_PAM
    char **tmp;
    char buf[PATH_MAX];
+   char *user;
+   ckc = ck_connector_new();
+   if( ckc != NULL )
+    {
+
+        user = g_key_file_get_string(config, "base", "autologin", 0);
+        struct passwd *pw;
+        pw = getpwnam(user);
+        DBusError error;
+        char x[256], *d, *n;
+        sprintf(x, "/dev/tty%d", tty);
+        dbus_error_init(&error);
+        d = x; n = getenv("DISPLAY");
+        if( ck_connector_open_session_with_parameters(ckc, &error,
+                                                      "unix-user", &pw->pw_uid,
+                                                      "display-device", &d,
+                                                      "x11-display-device", &d,
+                                                      "x11-display", &n,
+                                                      NULL) )
+            setenv("XDG_SESSION_COOKIE", ck_connector_get_cookie(ckc), 1);
+    }
+
+
+
    pid_t pid;
    pid = fork();
    if (pid == 0)
