diff -Nuar elsa-0.0_20110218.orig/data/themes/default.edc elsa-0.0_20110218/data/themes/default.edc
--- elsa-0.0_20110218.orig/data/themes/default.edc	2011-02-23 20:01:37.886000010 +0200
+++ elsa-0.0_20110218/data/themes/default.edc	2011-02-23 20:06:26.389000010 +0200
@@ -215,7 +215,7 @@
                rel2.relative: 0.5 0.0;
                rel2.offset: 0 50;
                text {
-                  text: "Welcome";
+                  text: "hoşgeldiniz";
                   font: "Sans";
                   size: 18;
                   min: 1 1;
@@ -238,7 +238,7 @@
                rel2.to: "box_login";
                rel2.relative: 0.5 1.0;
                text {
-                  text: "Error on login ! Maybee wrong password ?";
+                  text: "Girişte Hata! Parolayı yanlış girmiş olabilirsiniz...";
                   font: "Sans";
                   size: 18;
                   min: 1 1;
@@ -315,7 +315,7 @@
                text {
                   font: "Sans";
                   size: 18;
-                  text: "Login :";
+                  text: "Kullanıcı :";
                   min: 1 1;
                }
             }
@@ -380,7 +380,7 @@
                text {
                   font: "Sans";
                   size: 18;
-                  text: "Password :";
+                  text: "Parola :";
                   min: 1 1;
                }
             }
