Index: enlightenment-0.16.999.49539/data/other/Makefile.am
===================================================================
--- enlightenment-0.16.999.49539.orig/data/other/Makefile.am
+++ enlightenment-0.16.999.49539/data/other/Makefile.am
@@ -1,9 +1,7 @@
 MAINTAINERCLEANFILES = Makefile.in
 filesdir = $(datadir)/enlightenment/data/other
 files_DATA = \
-desktop_files.tar.gz \
 desktop_order.tar.gz \
-efm_favorites.tar.gz \
 icon_example.tar.gz
 
 EXTRA_DIST = $(files_DATA)
