Index: config/standard/e.src
===================================================================
--- config/standard/e.src	(revision 58042)
+++ config/standard/e.src	(working copy)
@@ -520,6 +520,15 @@
       value "priority" int: 0;
     }
   }
+  group "modules" list {·
+    group "E_Config_Module" struct {
+      value "name" string: "systray";
+      value "enabled" uchar: 1;
+      value "delayed" uchar: 1;
+      value "priority" int: 0;
+    }
+  }
+
   group "themes" list {
     group "E_Config_Theme" struct {
       value "category" string: "theme";
