diff -Nuar e.orig/config/standard/e.src e/config/standard/e.src
--- e.orig/config/standard/e.src	2011-03-17 19:08:36.000000000 +0200
+++ e/config/standard/e.src	2011-03-31 17:17:31.134000013 +0300
@@ -32,6 +32,15 @@
   value "evas_engine_zone" int: 0;
   value "use_composite" int: 0;
   value "language" string: "en_US.UTF-8";
+
+  group "modules" list {·
+    group "E_Config_Module" struct {
+      value "name" string: "systray";
+      value "enabled" uchar: 1;
+      value "delayed" uchar: 1;
+      value "priority" int: 0;
+    }
+  }
   group "modules" list {
     group "E_Config_Module" struct {
       value "name" string: "mixer";
