diff -Nuar enlightenment-0.16.999_52995.orig/config/standard/e.src enlightenment-0.16.999_52995/config/standard/e.src
--- enlightenment-0.16.999_52995.orig/config/standard/e.src	2010-10-03 22:06:05.000000000 +0300
+++ enlightenment-0.16.999_52995/config/standard/e.src	2011-02-23 17:09:12.406000004 +0200
@@ -2,6 +2,7 @@
   value "config_version" int: 65847;
   value "show_splash" int: 1;
   value "init_default_theme" string: "default.edj";
+  value "desktop_default_background" string: "pardus-2011_background.edj";
   value "desktop_default_name" string: "Desktop %i, %i";
   value "menus_scroll_speed" double: 1000.0000000000000000000000000;
   value "menus_fast_mouse_move_threshhold" double: 300.0000000000000000000000000;
