diff -Nuar lxinput-0.1.1.orig/data/lxinput.desktop lxinput-0.1.1/data/lxinput.desktop
--- lxinput-0.1.1.orig/data/lxinput.desktop	2010-09-25 13:58:33.367013897 +0300
+++ lxinput-0.1.1/data/lxinput.desktop	2010-09-25 13:58:53.101766285 +0300
@@ -2,11 +2,13 @@
 Type=Application
 Name=Keyboard and Mouse
 Name[ja]=キーボードとマウス
+Name[tr]=Klavye ve Fare
 Name[zh_TW]=鍵盤及滑鼠
 Icon=input-keyboard
 Exec=lxinput
 Comment=Configure keyboard, mouse, and other input devices
 Comment[ja]=キーボードやマウス、その他の入力機器の設定
+Comment[tr]=Klavye, fare ve diğer cihazlarınızı yapılandırın
 Comment[zh_TW]=設定鍵盤、滑鼠、及其他輸入裝置
 StartupNotify=true
 Terminal=false
