--- config/gdm.conf.in.orig	2009-03-19 06:59:27.000000000 +0200
+++ config/gdm.conf.in	2009-08-06 21:00:05.181949881 +0300
@@ -68,7 +68,7 @@
 
 # The greeter for attached (non-xdmcp) logins.  Change gdmlogin to gdmgreeter
 # to get the new graphical greeter.
-#Greeter=@libexecdir@/gdmlogin
+Greeter=@libexecdir@/gdmgreeter
 
 # The greeter for xdmcp logins, usually you want a less graphically intensive
 # greeter here so it's better to leave this with gdmlogin
@@ -163,9 +163,9 @@
 # really a PATH style variable since 2.4.4.2 to allow actual interoperability
 # with KDM.  Note that <dmconfdir>/Sessions is there for backwards
 # compatibility reasons with 2.4.4.x.
-#SessionDesktopDir=/etc/X11/sessions/:@dmconfdir@/Sessions/:@datadir@/gdm/BuiltInSessions/:@datadir@/xsessions/
+SessionDesktopDir=/etc/X11/sessions/:@dmconfdir@/Sessions/:@datadir@/gdm/BuiltInSessions/:@datadir@/xsessions/
 # This is the default .desktop session.  One of the ones in SessionDesktopDir
-#DefaultSession=gnome.desktop
+DefaultSession=gnome.desktop
 # Better leave this blank and HOME will be used.  You can use syntax ~/ below
 # to indicate home directory of the user.  You can also set this to something
 # like /tmp if you don't want the authorizations to be in home directories.
@@ -242,7 +242,7 @@
 [security]
 # Allow root to login.  It makes sense to turn this off for kiosk use, when
 # you want to minimize the possibility of break in.
-AllowRoot=true
+AllowRoot=false
 # Allow login as root via XDMCP.  This value will be overridden and set to
 # false if the /etc/default/login file exists and contains
 # "CONSOLE=/dev/login", and set to true if the /etc/default/login file exists
@@ -397,7 +397,7 @@
 # User ID's less than the MinimalUID value will not be included in the face
 # browser or in the gdmselection list for Automatic/Timed login.  They will not
 # be displayed regardless of the settings for Include and Exclude.
-#MinimalUID=100
+MinimalUID=1000
 # Users listed in Include will be included in the face browser and in the
 # gdmsetup selection list for Automatic/Timed login.  Users should be separated
 # by commas.
@@ -414,7 +414,7 @@
 # large numbers of users and this feature should not be used in such
 # environments.  The setting of IncludeAll does nothing if Include is set to a
 # non-empty value.
-#IncludeAll=false
+IncludeAll=true
 # If user or user.png exists in this dir it will be used as his picture.
 #GlobalFaceDir=@datadir@/pixmaps/faces/
 
@@ -470,8 +470,8 @@
 # The Standard greeter (gdmlogin) uses BackgroundColor as the background
 # color, while the themed greeter (gdmgreeter) uses GraphicalThemedColor
 # as the background color.
-BackgroundColor=#76848F
-GraphicalThemedColor=#76848F
+BackgroundColor=#000000
+GraphicalThemedColor=#000000
 # XDMCP session should only get a color, this is the sanest setting since you
 # don't want to take up too much bandwidth
 #BackgroundRemoteOnlyColor=true
@@ -492,7 +492,7 @@
 # Show the Failsafe sessions.  These are much MUCH nicer (focus for xterm for
 # example) and more failsafe then those supplied by scripts so distros should
 # use this rather then just running an xterm from a script.
-#ShowGnomeFailsafeSession=true
+ShowGnomeFailsafeSession=false
 #ShowXtermFailsafeSession=true
 # Normally there is a session type called 'Last' that is shown which refers to
 # the last session the user used.  If off, we will be in 'switchdesk' mode
@@ -509,7 +509,7 @@
 # list then provide a list that is delimited by /: to the GraphicalThemes
 # key and set GraphicalThemeRand to true.  Otherwise use GraphicalTheme
 # and specify just one theme.
-#GraphicalTheme=circles
+GraphicalTheme=pardus-air
 #GraphicalThemes=circles/:happygnome
 GraphicalThemeDir=@datadir@/gdm/themes/
 GraphicalThemeRand=false
@@ -525,7 +525,7 @@
 # user input.  If SoundOnLogin is a file and the greeter finds the 'play'
 # executable (see daemon/SoundProgram) it will play that file instead of just
 # beeping.
-#SoundOnLogin=true
+SoundOnLogin=false
 #SoundOnLoginFile=
 # If SoundOnLoginSuccess, then the greeter will play a sound (as above) when a
 # user successfully logs in.
