diff -Nuar disk-manager-3.0.0/po/az.po disk-manager/po/az.po
--- disk-manager-3.0.0/po/az.po	2010-08-09 09:37:31.000000000 +0300
+++ disk-manager/po/az.po	2011-02-23 13:59:11.291000049 +0200
@@ -46,6 +46,10 @@
 msgid "Your emails"
 msgstr "barish.kutsal@gmail.com"
 
+#: src/diskmanager/main.py:70 src/diskmanager/main.py:211 src/diskmanager/main.py:213 src/diskmanager/main.py:162 src/diskmanager/main.py:164 src/diskmanager/main.py:172 src/diskmanager/main.py:174
+msgid "Error"
+msgstr "Səhv"
+
 #: src/diskmanager/main.py:86
 msgid ""
 "There are no packages that provide backend for this application.\n"
diff -Nuar disk-manager-3.0.0/po/da.po disk-manager/po/da.po
--- disk-manager-3.0.0/po/da.po	2010-07-06 14:16:28.000000000 +0300
+++ disk-manager/po/da.po	2011-02-23 13:59:11.291000049 +0200
@@ -49,6 +49,10 @@
 msgid "Your names"
 msgstr ""
 
+#: src/diskmanager/main.py:70 src/diskmanager/main.py:211 src/diskmanager/main.py:213 src/diskmanager/main.py:162 src/diskmanager/main.py:164 src/diskmanager/main.py:172 src/diskmanager/main.py:174
+msgid "Error"
+msgstr "Fejl"
+
 #: src/diskmanager/main.py:86
 msgid ""
 "There are no packages that provide backend for this application.\n"
diff -Nuar disk-manager-3.0.0/po/de.po disk-manager/po/de.po
--- disk-manager-3.0.0/po/de.po	2011-01-04 22:52:32.000000000 +0200
+++ disk-manager/po/de.po	2011-02-23 13:59:11.168000049 +0200
@@ -46,6 +46,10 @@
 msgid "Your emails"
 msgstr "sw@pardususer.de"
 
+#: src/diskmanager/main.py:70 src/diskmanager/main.py:211 src/diskmanager/main.py:213 src/diskmanager/main.py:162 src/diskmanager/main.py:164 src/diskmanager/main.py:172 src/diskmanager/main.py:174
+msgid "Error"
+msgstr "Fehler"
+
 #: src/diskmanager/main.py:86
 msgid ""
 "There are no packages that provide backend for this application.\n"
diff -Nuar disk-manager-3.0.0/po/es.po disk-manager/po/es.po
--- disk-manager-3.0.0/po/es.po	2010-12-20 21:01:31.000000000 +0200
+++ disk-manager/po/es.po	2011-02-23 13:59:11.257000048 +0200
@@ -52,6 +52,10 @@
 msgid "Your emails"
 msgstr "click3d (at) linuxmail.org"
 
+#: src/diskmanager/main.py:70 src/diskmanager/main.py:211 src/diskmanager/main.py:213 src/diskmanager/main.py:162 src/diskmanager/main.py:164 src/diskmanager/main.py:172 src/diskmanager/main.py:174
+msgid "Error"
+msgstr "Error"
+
 #: src/diskmanager/main.py:86
 msgid ""
 "There are no packages that provide backend for this application.\n"
diff -Nuar disk-manager-3.0.0/po/fr.po disk-manager/po/fr.po
--- disk-manager-3.0.0/po/fr.po	2010-08-09 09:37:31.000000000 +0300
+++ disk-manager/po/fr.po	2011-02-23 13:59:11.277000048 +0200
@@ -47,6 +47,10 @@
 msgid "Your emails"
 msgstr "Votre courriel"
 
+#: src/diskmanager/main.py:70 src/diskmanager/main.py:211 src/diskmanager/main.py:213 src/diskmanager/main.py:162 src/diskmanager/main.py:164 src/diskmanager/main.py:172 src/diskmanager/main.py:174
+msgid "Error"
+msgstr "Erreur"
+
 #: src/diskmanager/main.py:86
 msgid ""
 "There are no packages that provide backend for this application.\n"
diff -Nuar disk-manager-3.0.0/po/hu.po disk-manager/po/hu.po
--- disk-manager-3.0.0/po/hu.po	2010-11-29 23:17:53.000000000 +0200
+++ disk-manager/po/hu.po	2011-02-23 13:59:11.290000049 +0200
@@ -47,6 +47,10 @@
 msgid "Your emails"
 msgstr "bicskas9323@gmail.com,fmihailov@gmail.com, sipiatti@gmail.com"
 
+#: src/diskmanager/main.py:70 src/diskmanager/main.py:211 src/diskmanager/main.py:213 src/diskmanager/main.py:162 src/diskmanager/main.py:164 src/diskmanager/main.py:172 src/diskmanager/main.py:174
+msgid "Error"
+msgstr "Hiba"
+
 #: src/diskmanager/main.py:86
 msgid ""
 "There are no packages that provide backend for this application.\n"
diff -Nuar disk-manager-3.0.0/po/it.po disk-manager/po/it.po
--- disk-manager-3.0.0/po/it.po	2010-07-06 14:16:28.000000000 +0300
+++ disk-manager/po/it.po	2011-02-23 13:59:11.195000049 +0200
@@ -49,6 +49,10 @@
 msgid "Your emails"
 msgstr "edivad95@gmail.com"
 
+#: src/diskmanager/main.py:70 src/diskmanager/main.py:211 src/diskmanager/main.py:213 src/diskmanager/main.py:162 src/diskmanager/main.py:164 src/diskmanager/main.py:172 src/diskmanager/main.py:174
+msgid "Error"
+msgstr "Errore"
+
 #: src/diskmanager/main.py:86
 msgid ""
 "There are no packages that provide backend for this application.\n"
diff -Nuar disk-manager-3.0.0/po/nl.po disk-manager/po/nl.po
--- disk-manager-3.0.0/po/nl.po	2010-07-06 14:16:28.000000000 +0300
+++ disk-manager/po/nl.po	2011-02-23 13:59:11.195000049 +0200
@@ -48,6 +48,10 @@
 msgid "Your emails"
 msgstr "atolboo@gmail.com"
 
+#: src/diskmanager/main.py:70 src/diskmanager/main.py:211 src/diskmanager/main.py:213 src/diskmanager/main.py:162 src/diskmanager/main.py:164 src/diskmanager/main.py:172 src/diskmanager/main.py:174
+msgid "Error"
+msgstr "Fout"
+
 #: src/diskmanager/main.py:86
 msgid ""
 "There are no packages that provide backend for this application.\n"
diff -Nuar disk-manager-3.0.0/po/pt_BR.po disk-manager/po/pt_BR.po
--- disk-manager-3.0.0/po/pt_BR.po	2010-12-26 17:47:21.000000000 +0200
+++ disk-manager/po/pt_BR.po	2011-02-23 13:59:11.023000050 +0200
@@ -46,6 +46,10 @@
 msgid "Your emails"
 msgstr "Seus emails"
 
+#: src/diskmanager/main.py:70 src/diskmanager/main.py:211 src/diskmanager/main.py:213 src/diskmanager/main.py:162 src/diskmanager/main.py:164 src/diskmanager/main.py:172 src/diskmanager/main.py:174
+msgid "Error"
+msgstr "Erro"
+
 #: src/diskmanager/main.py:86
 msgid ""
 "There are no packages that provide backend for this application.\n"
diff -Nuar disk-manager-3.0.0/po/ru.po disk-manager/po/ru.po
--- disk-manager-3.0.0/po/ru.po	2010-11-29 23:17:53.000000000 +0200
+++ disk-manager/po/ru.po	2011-02-23 13:59:11.268000048 +0200
@@ -46,6 +46,10 @@
 msgid "Your emails"
 msgstr "zag@vspu.ru, tribunal2000@gmail.com, neocrust@gmail.com"
 
+#: src/diskmanager/main.py:70 src/diskmanager/main.py:211 src/diskmanager/main.py:213 src/diskmanager/main.py:162 src/diskmanager/main.py:164 src/diskmanager/main.py:172 src/diskmanager/main.py:174
+msgid "Error"
+msgstr "Ошибка"
+
 #: src/diskmanager/main.py:86
 msgid ""
 "There are no packages that provide backend for this application.\n"
diff -Nuar disk-manager-3.0.0/po/sr.po disk-manager/po/sr.po
--- disk-manager-3.0.0/po/sr.po	2010-07-20 16:16:24.000000000 +0300
+++ disk-manager/po/sr.po	2011-02-23 13:59:11.246000048 +0200
@@ -48,6 +48,10 @@
 msgid "Your emails"
 msgstr "Kurucity@aim.com"
 
+#: src/diskmanager/main.py:70 src/diskmanager/main.py:211 src/diskmanager/main.py:213 src/diskmanager/main.py:162 src/diskmanager/main.py:164 src/diskmanager/main.py:172 src/diskmanager/main.py:174
+msgid "Error"
+msgstr "Грешка"
+
 #: src/diskmanager/main.py:86
 msgid ""
 "There are no packages that provide backend for this application.\n"
diff -Nuar disk-manager-3.0.0/po/sv.po disk-manager/po/sv.po
--- disk-manager-3.0.0/po/sv.po	2010-07-06 14:16:28.000000000 +0300
+++ disk-manager/po/sv.po	2011-02-23 13:59:11.292000049 +0200
@@ -47,6 +47,10 @@
 msgid "Your emails"
 msgstr "patrik@trixon.se"
 
+#: src/diskmanager/main.py:70 src/diskmanager/main.py:211 src/diskmanager/main.py:213 src/diskmanager/main.py:162 src/diskmanager/main.py:164 src/diskmanager/main.py:172 src/diskmanager/main.py:174
+msgid "Error"
+msgstr "Fel"
+
 #: src/diskmanager/main.py:86
 msgid ""
 "There are no packages that provide backend for this application.\n"
diff -Nuar disk-manager-3.0.0/po/tr.po disk-manager/po/tr.po
--- disk-manager-3.0.0/po/tr.po	2010-08-09 09:37:31.000000000 +0300
+++ disk-manager/po/tr.po	2011-02-23 13:59:11.185000049 +0200
@@ -50,6 +50,10 @@
 msgid "Your emails"
 msgstr "bahadir@pardus.org.tr, necmettin.begiter@gmail.com"
 
+#: src/diskmanager/main.py:70 src/diskmanager/main.py:211 src/diskmanager/main.py:213 src/diskmanager/main.py:162 src/diskmanager/main.py:164 src/diskmanager/main.py:172 src/diskmanager/main.py:174
+msgid "Error"
+msgstr "Hata"
+
 #: src/diskmanager/main.py:86
 msgid ""
 "There are no packages that provide backend for this application.\n"
diff -Nuar disk-manager-3.0.0/setup.py disk-manager/setup.py
--- disk-manager-3.0.0/setup.py	2010-12-17 14:07:20.000000000 +0200
+++ disk-manager/setup.py	2011-02-23 13:59:11.305000049 +0200
@@ -26,6 +26,13 @@
 from distutils.command.install import install
 
 PROJECT = about.appName
+FOR_KDE_4=False
+
+if 'kde4' in sys.argv:
+    sys.argv.remove('kde4')
+    FOR_KDE_4=True
+    print 'UI files will be created for KDE 4.. '
+
 
 def makeDirs(directory):
     if not os.path.exists(directory):
@@ -48,7 +55,10 @@
     # Collect UI files
     filelist = []
     for filename in glob.glob1("ui", "*.ui"):
-        os.system("pykde4uic -o ui/ui_%s.py ui/%s" % (filename.split(".")[0], filename))
+        if FOR_KDE_4:
+            os.system("pykde4uic -o ui/ui_%s.py ui/%s -g &s" % (filename.split(".")[0], filename, PROJECT))
+        else:
+            os.system("pyuic4 -o ui/ui_%s.py ui/%s -g %s" % (filename.split(".")[0], filename, PROJECT))
 
     # Collect headers for desktop files
     for filename in glob.glob("data/*.desktop.in"):
@@ -105,7 +115,10 @@
 
         print "Generating UIs..."
         for filename in glob.glob1("ui", "*.ui"):
-            os.system("pykde4uic -o build/diskmanager/ui_%s.py ui/%s" % (filename.split(".")[0], filename))
+            if FOR_KDE_4:
+                os.system("pykde4uic -o build/diskmanager/ui_%s.py ui/%s -g %s" % (filename.split(".")[0], filename, PROJECT))
+            else:
+                os.system("pyuic4 -o build/diskmanager/ui_%s.py ui/%s -g %s" % (filename.split(".")[0], filename, PROJECT))
 
         print "Generating RCs..."
         for filename in glob.glob1("data", "*.qrc"):
@@ -128,9 +141,9 @@
             bin_dir = "/usr/bin"
 
         locale_dir = os.path.join(root_dir, "locale")
-        apps_dir = os.path.join(root_dir, "applications/kde4")
-        services_dir = os.path.join(root_dir, "kde4/services")
-        project_dir = os.path.join(root_dir, "kde4/apps", PROJECT)
+        apps_dir = os.path.join(root_dir, "applications")
+        #services_dir = os.path.join(root_dir, "kde4/services")
+        project_dir = os.path.join(root_dir, PROJECT)
 
         # Make directories
         print "Making directories..."
@@ -138,13 +151,14 @@
         makeDirs(locale_dir)
         makeDirs(apps_dir)
         makeDirs(project_dir)
-        makeDirs(services_dir)
+        #makeDirs(services_dir)
 
         # Install desktop files
         print "Installing desktop files..."
-
-        shutil.copy("data/%s.desktop" % PROJECT, apps_dir)
-        shutil.copy("data/kcm_%s.desktop" % PROJECT, services_dir)
+        if FOR_KDE_4:
+            shutil.copy("data/kcm_%s.desktop" % PROJECT, apps_dir)
+        else:
+            shutil.copy("data/%s.desktop" % PROJECT, apps_dir)
         shutil.rmtree('build/data')
 
         # Install codes
@@ -192,14 +206,14 @@
         bin_dir = "/usr/bin"
 
         locale_dir = os.path.join(root_dir, "locale")
-        apps_dir = os.path.join(root_dir, "applications/kde4")
-        services_dir = os.path.join(root_dir, "kde4/services")
-        project_dir = os.path.join(root_dir, "kde4/apps", PROJECT)
+        apps_dir = os.path.join(root_dir, "applications")
+        #services_dir = os.path.join(root_dir, "kde4/services")
+        project_dir = os.path.join(root_dir, PROJECT)
 
         print 'Uninstalling ...'
         remove(project_dir)
         remove(apps_dir +"/%s.desktop" % PROJECT)
-        remove(services_dir +"/kcm_%s.desktop" % PROJECT)
+        #remove(services_dir +"/kcm_%s.desktop" % PROJECT)
         for filename in glob.glob1('po', '*.po'):
             lang = filename.rsplit(".", 1)[0]
             remove(os.path.join(locale_dir, "%s/LC_MESSAGES" % lang, "%s.mo" % PROJECT))
@@ -225,8 +239,8 @@
       description       = unicode(about.PACKAGE),
       license           = unicode('GPL'),
       author            = "Pardus Developers",
-      author_email      = about.bugEmail,
-      url               = about.homePage,
+      author_email      = "bug@pardus.org.tr",
+      url               = "http://www.pardus.org.tr/eng/projects",
       packages          = [''],
       package_dir       = {'': ''},
       data_files        = [],
diff -Nuar disk-manager-3.0.0/src/diskmanager/about.py disk-manager/src/diskmanager/about.py
--- disk-manager-3.0.0/src/diskmanager/about.py	2010-12-26 17:47:21.000000000 +0200
+++ disk-manager/src/diskmanager/about.py	2011-02-23 13:59:11.329000049 +0200
@@ -11,29 +11,35 @@
 # Please read the COPYING file.
 #
 
+import context as ctx
+
 PACKAGE = "Disk Manager"
+appName= "disk-manager"
+version="3.0.0"
+
+if ctx.Pds.session == ctx.pds.Kde4:
+
+    # PyKDE
+    from PyKDE4.kdecore import KAboutData, ki18n, ki18nc
 
-# PyKDE
-from PyKDE4.kdecore import KAboutData, ki18n, ki18nc
+    # Application Data
+    appName     = "disk-manager"
+    modName     = "diskmanager"
+    programName = ki18n("Disk Manager")
+    version     = "3.0.0"
+    description = ki18n("Disk Manager")
+    license     = KAboutData.License_GPL
+    copyright   = ki18n("(c) 2006-2010 TUBITAK/UEKAE")
+    text        = ki18n(None)
+    homePage    = "http://www.pardus.org.tr/eng/projects"
+    bugEmail    = "bugs@pardus.org.tr"
+    catalog     = appName
+    aboutData   = KAboutData(appName, catalog, programName, version, description, license, copyright, text, homePage, bugEmail)
 
-# Application Data
-appName     = "disk-manager"
-modName     = "diskmanager"
-programName = ki18n("Disk Manager")
-version     = "3.0.0"
-description = ki18n("Disk Manager")
-license     = KAboutData.License_GPL
-copyright   = ki18n("(c) 2006-2010 TUBITAK/UEKAE")
-text        = ki18n(None)
-homePage    = "http://www.pardus.org.tr/eng/projects"
-bugEmail    = "bugs@pardus.org.tr"
-catalog     = appName
-aboutData   = KAboutData(appName, catalog, programName, version, description, license, copyright, text, homePage, bugEmail)
-
-# Author(s)
-aboutData.addAuthor(ki18n("Gökmen Göksel"), ki18n("Current Maintainer"))
-aboutData.addAuthor(ki18n("Bahadır Kandemir"), ki18n("First Developer"))
-aboutData.setTranslator(ki18nc("NAME OF TRANSLATORS", "Your names"), ki18nc("EMAIL OF TRANSLATORS", "Your emails"))
+    # Author(s)
+    aboutData.addAuthor(ki18n("Gökmen Göksel"), ki18n("Current Maintainer"))
+    aboutData.addAuthor(ki18n("Bahadır Kandemir"), ki18n("First Developer"))
+    aboutData.setTranslator(ki18nc("NAME OF TRANSLATORS", "Your names"), ki18nc("EMAIL OF TRANSLATORS", "Your emails"))
 
-# Use this if icon name is different than appName
-aboutData.setProgramIconName("drive-harddisk")
+    # Use this if icon name is different than appName
+    aboutData.setProgramIconName("drive-harddisk")
diff -Nuar disk-manager-3.0.0/src/diskmanager/config.py disk-manager/src/diskmanager/config.py
--- disk-manager-3.0.0/src/diskmanager/config.py	2010-07-06 14:16:28.000000000 +0300
+++ disk-manager/src/diskmanager/config.py	2011-02-23 13:59:11.325000049 +0200
@@ -11,7 +11,6 @@
 # Please read the COPYING file.
 #
 
-from PyKDE4.kdecore import i18n
 
 # FS types and options
 FS_TYPES = {
diff -Nuar disk-manager-3.0.0/src/diskmanager/context.py disk-manager/src/diskmanager/context.py
--- disk-manager-3.0.0/src/diskmanager/context.py	1970-01-01 02:00:00.000000000 +0200
+++ disk-manager/src/diskmanager/context.py	2011-02-23 13:59:11.329000049 +0200
@@ -0,0 +1,29 @@
+#!/usr/bin/python
+# -*- coding: utf-8 -*-
+
+import pds
+import traceback
+from time import time
+from pds.qiconloader import QIconLoader
+
+Pds = pds.Pds('disk-manager', debug = True)
+# Force to use Default Session for testing
+Pds.session = pds.DefaultDe
+# print 'Current session is : %s %s' % (Pds.session.Name, Pds.session.Version)
+
+i18n = Pds.i18n
+KIconLoader = QIconLoader(Pds)
+KIcon = KIconLoader.icon
+
+time_counter = 0
+start_time = time()
+last_time = time()
+
+def _time():
+    global last_time, time_counter
+    trace = list(traceback.extract_stack())
+    diff = time() - start_time
+    print ('%s ::: %s:%s' % (time_counter, trace[-2][0].split('/')[-1], trace[-2][1])), diff, diff - last_time
+    last_time = diff
+    time_counter += 1
+
diff -Nuar disk-manager-3.0.0/src/diskmanager/edit.py disk-manager/src/diskmanager/edit.py
--- disk-manager-3.0.0/src/diskmanager/edit.py	2010-07-06 14:16:28.000000000 +0300
+++ disk-manager/src/diskmanager/edit.py	2011-02-23 13:59:11.329000049 +0200
@@ -15,10 +15,6 @@
 from PyQt4 import QtCore
 from PyQt4 import QtGui
 
-# PyKDE
-from PyKDE4 import kdeui
-from PyKDE4 import kdecore
-
 # UI
 from diskmanager.ui_edit import Ui_EditWidget
 
diff -Nuar disk-manager-3.0.0/src/diskmanager/item.py disk-manager/src/diskmanager/item.py
--- disk-manager-3.0.0/src/diskmanager/item.py	2010-07-06 14:16:28.000000000 +0300
+++ disk-manager/src/diskmanager/item.py	2011-02-23 13:59:11.329000049 +0200
@@ -15,8 +15,13 @@
 from PyQt4 import QtCore
 from PyQt4 import QtGui
 
-# PyKDE
-from PyKDE4.kdeui import KIcon
+#Pds Context Stuff
+import context as ctx
+if ctx.Pds.session == ctx.pds.Kde4:
+    # PyKDE
+    from PyKDE4.kdeui import KIcon
+else:
+    from context import *
 
 # UI
 from diskmanager.ui_item import Ui_ItemWidget
diff -Nuar disk-manager-3.0.0/src/diskmanager/kcmodule.py disk-manager/src/diskmanager/kcmodule.py
--- disk-manager-3.0.0/src/diskmanager/kcmodule.py	1970-01-01 02:00:00.000000000 +0200
+++ disk-manager/src/diskmanager/kcmodule.py	2011-02-23 13:59:11.329000049 +0200
@@ -0,0 +1,15 @@
+#!/usr/bin/python
+# -*- coding: utf-8 -*-
+
+class Module(KCModule):
+    def __init__(self, component_data, parent):
+        KCModule.__init__(self, component_data, parent)
+
+        KGlobal.locale().insertCatalog(catalog)
+
+        if not dbus.get_default_main_loop():
+            from dbus.mainloop.qt import DBusQtMainLoop
+            DBusQtMainLoop(set_as_default=True)
+
+        MainWidget(self, embed=True)
+
diff -Nuar disk-manager-3.0.0/src/diskmanager/main.py disk-manager/src/diskmanager/main.py
--- disk-manager-3.0.0/src/diskmanager/main.py	2010-12-26 17:47:21.000000000 +0200
+++ disk-manager/src/diskmanager/main.py	2011-02-23 14:12:28.851000050 +0200
@@ -13,11 +13,10 @@
 
 # PyQt
 from PyQt4 import QtCore
+from PyQt4.QtGui import QMessageBox,QIcon
 from PyQt4 import QtGui
-
-# PyKDE
-from PyKDE4 import kdeui
-from PyKDE4 import kdecore
+import context as ctx
+from diskmanager.context import *
 
 # UI
 from diskmanager.ui_main import Ui_MainWidget
@@ -37,6 +36,7 @@
 # Edit widget
 from diskmanager.pagedialog import PageDialog
 
+
 class MainWidget(QtGui.QWidget, Ui_MainWidget):
     def __init__(self, parent, embed=False):
         QtGui.QWidget.__init__(self, parent)
@@ -63,8 +63,8 @@
     def checkBackend(self):
         """ Check if there are packages that provide required backend. """
         if not len(self.iface.getPackages()):
-            kdeui.KMessageBox.error(self, \
-                    kdecore.i18n("There are no packages that provide " \
+            QMessageBox.critical(self, i18n("Error"), \
+                    i18n("There are no packages that provide " \
                                  "backend for this application.\n" \
                                  "Please make sure that packages " \
                                  "are installed and configured correctly."))
@@ -93,9 +93,9 @@
     def addItem(self, id_, name="", description="", mounted=False):
         """ Adds an item to list. """
         if mounted:
-            icon = kdeui.KIcon("drive-harddisk", None, ["emblem-mounted"])
+            icon = QIcon(KIconLoader.loadOverlayed('drive-harddisk', ["emblem-mounted"], 32))
         else:
-            icon = kdeui.KIcon("drive-harddisk")
+            icon = KIcon("drive-harddisk")
 
         type_ = "disk"
 
@@ -135,8 +135,7 @@
                     parts.sort()
                     for part in parts:
                         if part in self.mounted_devices:
-                            description = kdecore.i18n("Mounted at %1", \
-                                    self.mounted_devices[part])
+                            description =i18n("Mounted at %1" , self.mounted_devices[part])
                             self.addItem(part, part, description, True)
                         else:
                             self.addItem(part, part, "")
@@ -156,10 +155,9 @@
                 self.iface.mount(widget.getId(), path)
             except Exception, e:
                 if "Comar.PolicyKit" in e._dbus_error_name:
-                    kdeui.KMessageBox.error(self,
-                            kdecore.i18n("Access denied."))
+                    QMessageBox.critical(self, i18n("Error"),  i18n("Access denied."))
                 else:
-                    kdeui.KMessageBox.error(self, unicode(e))
+                    QMessageBox.critical(self, i18n("Error"), unicode(e))
                 widget.setState(False)
                 return
         elif state == QtCore.Qt.Unchecked:
@@ -167,10 +165,9 @@
                 self.iface.umount(widget.getId())
             except Exception, e:
                 if "Comar.PolicyKit" in e._dbus_error_name:
-                    kdeui.KMessageBox.error(self,
-                            kdecore.i18n("Access denied."))
+                    QMessageBox.critical(self, i18n("Error"), i18n("Access denied."))
                 else:
-                    kdeui.KMessageBox.error(self, unicode(e))
+                    QMessageBox.critical(self, i18n("Error"), unicode(e))
                 widget.setState(True)
                 return
         self.buildItemList()
@@ -207,9 +204,9 @@
                     self.iface.removeEntry(device)
             except Exception, e:
                 if "Comar.PolicyKit" in e._dbus_error_name:
-                    kdeui.KMessageBox.error(self, kdecore.i18n("Access denied."))
+                    QMessageBox.critical(self, i18n("Error"),  i18n("Access denied."))
                 else:
-                    kdeui.KMessageBox.error(self, unicode(e))
+                    QMessageBox.critical(self, i18n("Error"),  unicode(e))
 
     def slotItemDelete(self):
         """ Delete button clicked. """
diff -Nuar disk-manager-3.0.0/src/diskmanager/pagedialog.py disk-manager/src/diskmanager/pagedialog.py
--- disk-manager-3.0.0/src/diskmanager/pagedialog.py	2010-07-06 14:16:28.000000000 +0300
+++ disk-manager/src/diskmanager/pagedialog.py	2011-02-23 13:59:11.329000049 +0200
@@ -15,24 +15,52 @@
 from PyQt4 import QtCore
 from PyQt4 import QtGui
 
-# PyKDE
-from PyKDE4 import kdeui
-from PyKDE4 import kdecore
+#Pds Stuff
+from diskmanager.context import *
+import context as ctx
 
 # Edit widget
 from diskmanager.edit import EditWidget
 
-
-class PageDialog(kdeui.KPageDialog):
-    def __init__(self, parent):
-        kdeui.KPageDialog.__init__(self, parent)
-
-        self.setFaceType(kdeui.KPageDialog.Tabbed)
-        self.setCaption(kdecore.i18n("Settings"))
-
-        self.page_widget = EditWidget(self)
-        self.page_item = kdeui.KPageWidgetItem(self.page_widget, kdecore.i18n("Settings"))
-
-        self.addPage(self.page_item)
-
-        self.edit = self.page_widget
+if ctx.Pds.session == ctx.pds.Kde4:
+    #PyKDE
+    from PyKDE4 import kdeui
+    from PyKDE4 import kdecore
+
+    class PageDialog(kdeui.KPageDialog):
+        def __init__(self, parent):
+            kdeui.KPageDialog.__init__(self, parent)
+
+            self.setFaceType(kdeui.KPageDialog.Tabbed)
+            self.setCaption(i18n("Settings"))
+
+            self.page_widget = EditWidget(self)
+            self.page_item = kdeui.KPageWidgetItem(self.page_widget, i18n("Settings"))
+
+            self.addPage(self.page_item)
+            self.edit = self.page_widget
+
+else:
+    class PageDialog(QtGui.QDialog):
+        def __init__(self,parent):
+            QtGui.QDialog.__init__(self,parent)
+            self.setWindowTitle(i18n("Settings"))
+            self.resize(548,180)
+            self.page_widget = EditWidget(self)
+            self.tab=QtGui.QTabWidget(self)
+            self.tab.addTab(self.page_widget,i18n("Settings"))
+
+            #Buttons
+            self.buttonBox = QtGui.QDialogButtonBox(self)
+            self.buttonBox.setGeometry(QtCore.QRect(4, 152, 540, 25))
+            self.buttonBox.setStandardButtons(QtGui.QDialogButtonBox.Cancel|QtGui.QDialogButtonBox.Ok)
+            self.layout=QtGui.QVBoxLayout(self)
+            self.layout.addWidget(self.tab)
+            self.layout.addWidget(self.buttonBox)
+            self.buttonBox.setObjectName(i18n("buttonBox"))
+
+            #SIGNAL
+            QtCore.QObject.connect(self.buttonBox, QtCore.SIGNAL(i18n("accepted()")), self.accept)
+            QtCore.QObject.connect(self.buttonBox, QtCore.SIGNAL(i18n("rejected()")), self.reject)
+            QtCore.QMetaObject.connectSlotsByName(self)
+            self.edit=self.page_widget
diff -Nuar disk-manager-3.0.0/src/disk-manager.py disk-manager/src/disk-manager.py
--- disk-manager-3.0.0/src/disk-manager.py	2010-07-06 14:16:28.000000000 +0300
+++ disk-manager/src/disk-manager.py	2011-02-23 13:59:11.324000049 +0200
@@ -10,23 +10,24 @@
 #
 # Please read the COPYING file.
 #
-
+#System
 import sys
 import dbus
+#Pds Stuff
+import diskmanager.context as ctx
+from diskmanager.context import *
+
+#Qt Stuff
+from PyQt4.QtGui import *
+from PyQt4.QtCore import *
 
-from PyQt4 import QtGui
-from PyQt4 import QtCore
-
-from PyKDE4.kdeui import KMainWindow, KApplication, KCModule, KIcon
-from PyKDE4.kdecore import KCmdLineArgs, KGlobal
 
-from diskmanager.about import aboutData, catalog
 from diskmanager.main import MainWidget
 
 
-class MainWindow(KMainWindow):
+class MainWindow(QMainWindow):
     def __init__(self, parent=None):
-        KMainWindow.__init__(self, parent)
+        QMainWindow.__init__(self, parent)
         widget = MainWidget(self)
         self.resize(widget.size())
         self.setCentralWidget(widget)
@@ -34,31 +35,50 @@
 
 if __name__ == "__main__":
 
-    KCmdLineArgs.init(sys.argv, aboutData)
-    app = KApplication()
-
+    #DBUS MainLoop
     if not dbus.get_default_main_loop():
         from dbus.mainloop.qt import DBusQtMainLoop
         DBusQtMainLoop(set_as_default=True)
 
-    window = MainWindow()
-    window.show()
-
-    app.exec_()
-
-
-class Module(KCModule):
-    def __init__(self, component_data, parent):
-        KCModule.__init__(self, component_data, parent)
-
-        KGlobal.locale().insertCatalog(catalog)
-
-        if not dbus.get_default_main_loop():
-            from dbus.mainloop.qt import DBusQtMainLoop
-            DBusQtMainLoop(set_as_default=True)
-
-        MainWidget(self, embed=True)
+    if ctx.Pds.session == ctx.pds.Kde4:
+        #PyKDE4 Stuff
+        from PyKDE4.kdeui import *
+        from PyKDE4.kdecore import *
+
+        # Application Stuff
+        from diskmanager.about import aboutData, catalog
+
+        # Set Command-line arguments
+        KCmdLineArgs.init(sys.argv, aboutData)
+
+        # Create a Kapplitcation instance
+        app = KApplication()
+
+        # Create Main Widget
+        window = MainWindow()
+        window.show()
+
+        # Run the application
+        app.exec_()
+    else:
+        import gettext
+        __trans = gettext.translation('disk-manager', fallback=True)
+        i18n = __trans.ugettext
+        #Pds Stuff
+        from pds.quniqueapp import QUniqueApplication
+
+        #Create a QUniqueApp instance
+        app = QUniqueApplication(sys.argv, catalog="disk-manager")
+
+        #Create Main Widget
+        window = MainWindow()
+        window.show()
+        window.resize(550, 484)
+        window.setWindowTitle(i18n("Disk Manager"))
+        window.setWindowIcon(KIcon("drive-harddisk"))
 
+        #Run the application
+        app.exec_()
 
 def CreatePlugin(widget_parent, parent, component_data):
     return Module(component_data, parent)
