--- gpicview-0.2.1.orig/gpicview.desktop.in
+++ gpicview-0.2.1/gpicview.desktop.in
@@ -15,6 +15,7 @@ Name[ja]=画像ビューワ
 Name[pl]=Przeglądarka obrazów
 Name[zh_CN]=图片查看器
 Name[zh_TW]=圖片檢視器
+Name[tr]=Resim Önizleyicisi
 GenericName=Image Viewer
 GenericName[et]=Pildivaatur
 GenericName[fi]=Kuvankatselija
@@ -23,6 +24,7 @@ GenericName[ja]=画像ビューワ
 GenericName[pl]=Przeglądarka obrazów
 GenericName[zh_CN]=图片查看器
 GenericName[zh_TW]=圖片檢視器
+GenericName[tr]=Resim Önizleyicisi
 Comment=View your images easily
 Comment[et]=Vaata pilte veel lihtsamalt
 Comment[fi]=Katso kuviasi helposti
@@ -30,4 +32,5 @@ Comment[fr]=Visionner des images facilem
 Comment[ja]=簡単に画像を見ることができます
 Comment[pl]=Wyświetla obrazy
 Comment[zh_CN]=轻松查看数码相片和图片
-Comment[zh_TW]=輕鬆檢視數位相片和圖片
\ No newline at end of file
+Comment[zh_TW]=輕鬆檢視數位相片和圖片
+Comment[tr]=Resimlerinize kolayca bakın

