Index: src/burn-extension.c
===================================================================
--- src/burn-extension.c
+++ src/burn-extension.c
@@ -129,12 +129,12 @@
                             &error)) {
 
 
-                dialog = gtk_message_dialog (NULL, GTK_DIALOG_MODAL, GTK_MESSAGE_WARNING,
-					     GTK_BUTTONS_OK, _("Unable to launch the cd burner application"));
+                dialog = gtk_message_dialog_new (NULL, GTK_DIALOG_MODAL, GTK_MESSAGE_WARNING,
+						 GTK_BUTTONS_OK, _("Unable to launch the cd burner application"));
 
-                gtk_message_dialog_format_secondary_text (dialog, "%s", error->message);
+                gtk_message_dialog_format_secondary_text (GTK_MESSAGE_DIALOG (dialog), "%s", error->message);
 
-                gtk_dialog_run (dialog);
+                gtk_dialog_run (GTK_DIALOG (dialog));
                 gtk_widget_destroy (dialog);
 
                 g_error_free (error);
