diff -Naur envision-0.0.1.orig//data/desktop/envision.desktop envision-0.0.1/data/desktop/envision.desktop
--- envision-0.0.1.orig//data/desktop/envision.desktop	2011-02-22 17:41:32.826000000 +0200
+++ envision-0.0.1/data/desktop/envision.desktop	2011-02-22 17:42:37.794000003 +0200
@@ -5,6 +5,7 @@
 Comment[fr]=Lecteur E-PDF
 Comment[it]=Visualizzatore E-PDF
 Comment[pt]=Leitor de PDF
+Comment[tr]=E-PDF Okuyucu
 Exec=envision
 Icon=envision
 Terminal=false
