--- gnome-system-monitor.desktop.in.in 2008-04-30 21:21:50.000000000 +0300
@@ -8,7 +8,8 @@
 Terminal=false
 Type=Application
 StartupNotify=true
-Categories=GNOME;GTK;System;Monitor;
+Categories=GNOME;GTK;System;Settings;
+OnlyShowIn=GNOME
 X-GNOME-Bugzilla-Bugzilla=GNOME
 X-GNOME-Bugzilla-Product=system-monitor
 X-GNOME-Bugzilla-Component=general
--- gnome-system-monitor.desktop.in 2008-04-30 21:21:50.000000000 +0300
@@ -8,7 +8,8 @@
 Terminal=false
 Type=Application
 StartupNotify=true
-Categories=GNOME;GTK;System;Monitor;
+Categories=GNOME;GTK;System;Settings;
+OnlyShowIn=GNOME
 X-GNOME-Bugzilla-Bugzilla=GNOME
 X-GNOME-Bugzilla-Product=system-monitor
 X-GNOME-Bugzilla-Component=general
