--- gnome-screensaver-2.25.2/src/gs-theme-manager.c	2007-12-12 16:39:11.000000000 -0500
+++ hacked/src/gs-theme-manager.c	2009-02-21 00:10:06.249343027 -0500
@@ -75,18 +75,35 @@
 {
         int i;
 
-        for (i = 0; known_engine_locations [i]; i++){
-                char *path;
+	if (g_path_is_absolute (command)) {
+		char *dirname;
 
-                path = g_build_filename (known_engine_locations [i], command, NULL);
-
-                if (g_file_test (path, G_FILE_TEST_IS_EXECUTABLE)
-                    && ! g_file_test (path, G_FILE_TEST_IS_DIR)) {
-                        return path;
-                }
-
-                g_free (path);
-        }
+		dirname = g_path_get_dirname (command);
+		for (i = 0; known_engine_locations [i]; i++) {
+			if (strcmp (dirname, known_engine_locations [i]) == 0) {
+                		if (g_file_test (command, G_FILE_TEST_IS_EXECUTABLE)
+                    		    && ! g_file_test (command, G_FILE_TEST_IS_DIR)) {
+					g_free (dirname);
+                        		return g_strdup (command);
+                		}
+			}
+		}
+		g_free (dirname);
+	}
+	else {
+        	for (i = 0; known_engine_locations [i]; i++){
+                	char *path;
+
+                	path = g_build_filename (known_engine_locations [i], command, NULL);
+
+                	if (g_file_test (path, G_FILE_TEST_IS_EXECUTABLE)
+                    	    && ! g_file_test (path, G_FILE_TEST_IS_DIR)) {
+                        	return path;
+                	}
+
+                	g_free (path);
+        	}
+	}
 
         return NULL;
 }
