Index: pds/__init__.py
===================================================================
--- pds/__init__.py	(revision 36469)
+++ pds/__init__.py	(working copy)
@@ -146,7 +146,7 @@
     def session(self):
         if not self._session:
             env = getenv('DESKTOP_SESSION')
-            if env == 'default':
+            if env == 'default' or not env:
                 session = readfile('/etc/default/desktop', DefaultDe.Name)
                 env     = session.split('=')[1].strip()
             for de in Pds.SupportedDesktops:
