Index: pds/__init__.py
===================================================================
--- pds/__init__.py	(revision 36366)
+++ pds/__init__.py	(working copy)
@@ -30,7 +30,7 @@
 
 class Pds:
 
-    SupportedDesktops = (DefaultDe, Kde4, Kde3, Xfce, Enlightenment)
+    SupportedDesktops = (DefaultDe, Kde4, Kde3, Xfce, Enlightenment, LXDE)
 
     def __init__(self, catalogName='', debug=False):
         self._session           = None
Index: pds/environments.py
===================================================================
--- pds/environments.py	(revision 36366)
+++ pds/environments.py	(working copy)
@@ -74,3 +74,13 @@
     DefaultIconTheme     = 'hicolor'
     IconKey              = 'E_ICON_THEME'
 
+class LXDE(DefaultDe):
+    Name = 'LXDE'
+    Version = '0.5'
+    ConfigPath = '$HOME/.config/openbox'
+    ConfigFile = 'lxde-rc.xml'
+    ConfigType = 'xml'
+    DefaultIconTheme = 'oxygen'
+    IconKey = 'theme/name'
+
+
