diff -Nuar epdfview-0.1.7.orig//data/epdfview.desktop epdfview-0.1.7//data/epdfview.desktop
--- epdfview-0.1.7.orig//data/epdfview.desktop	2011-02-09 11:01:15.313000008 +0200
+++ epdfview-0.1.7//data/epdfview.desktop	2011-02-09 11:03:38.082000006 +0200
@@ -7,7 +7,7 @@
 GenericName=PDF Viewer
 GenericName[ca]=Visor PDF
 GenericName[es]=Visor PDF
-Icon=icon_epdfview-48
+Icon=/usr/share/epdfview/pixmaps/icon_epdfview-48.png
 Name=ePDFViewer
 Name[ca]=ePDFViewer
 Name[es]=ePDFViewer
