diff -Nuar kaptan-e17//src/kaptan/screens/scrPackage.py /home/engin/İndirilenler/kaptan-e17//src/kaptan/screens/scrPackage.py
--- kaptan-e17//src/kaptan/screens/scrPackage.py	2011-02-23 16:25:52.000000000 +0200
+++ /home/engin/İndirilenler/kaptan-e17//src/kaptan/screens/scrPackage.py	2011-02-24 00:51:51.863000001 +0200
@@ -20,7 +20,7 @@
 from kaptan.screens.ui_scrPackage import Ui_packageWidget
 
 import subprocess
-
+import os
 isUpdateOn = False
 
 class Widget(QtGui.QWidget, Screen):
@@ -34,9 +34,9 @@
         QtGui.QWidget.__init__(self,None)
         self.ui = Ui_packageWidget()
         self.ui.setupUi(self)
-        self.ui.checkBox.setChecked(True)
+        self.ui.checkBox.setChecked(False)
+        os.system("rm -rf /etc/xdg/autostart/kaptan.desktop")
 
-        
         self.connect(self.ui.checkBox,SIGNAL("stateChanged(int)"),self.addRepo)
 
     def addRepo(self):
@@ -50,7 +50,6 @@
             #self.__class__.screenSettings["summaryMessage"].update({"selectedRepo ": i18n("True")})
             #from pisi import api as addRepo
             #addRepo.add_repo("enlightment", "http://x86-64.comu.edu.tr/lxde/i686/pisi-index.xml.xz")
-
         else:
             pass
     def enableCheckTime(self):
