--- lxtask.desktop.in.orig	2010-11-11 01:09:06.561000042 +0200
+++ lxtask.desktop.in	2010-11-11 01:10:31.159000045 +0200
@@ -1,7 +1,10 @@
 [Desktop Entry]
-_Name=Task Manager
+_Name=LXDE Task Manager
+Name[tr]=LXDE için Görev Yöneticisi
 _Comment=Manage running processes
+Comment[tr]=Çalışan süreçleri yönet
 _GenericName=Process manager
+GenericName[tr]=İşlem yöneticisi
 Exec=lxtask
 Icon=utilities-system-monitor
 Terminal=false
